plugin/lspconfig.lua

```diff
@@ -110,20 +110,22 @@ end, {
 
 api.nvim_create_user_command('LspStop', function(info)
   local current_buf = vim.api.nvim_get_current_buf()
-  local server_id, force
+  local server_id, force, server_name
   local arguments = vim.split(info.args, '%s')
   for _, v in pairs(arguments) do
     if v == '++force' then
       force = true
     elseif v:find '^[0-9]+$' then
       server_id = v
+    else
+      server_name = v
     end
   end
 
   if not server_id then
     local servers_on_buffer = require('lspconfig.util').get_lsp_clients { bufnr = current_buf }
     for _, client in ipairs(servers_on_buffer) do
-      if client.attached_buffers[current_buf] then
+      if client.attached_buffers[current_buf] and (server_name and (server_name == client.config.name) or true) then
         client.stop(force)
       end
```

Comment:
if server_name is provided but not found, we should show a `config "foo" not found` message to indicate that. silent failure is usually a bad UX.

Reply:
make sense